src/main/java/core/basesyntax/operation/ReturnHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class ReturnHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getOperation() == FruitTransaction.Operation.RETURN) {

```

Comment:
The check for the operation type should not be the responsibility of the `apply` method. This responsibility should be in the class that uses the Strategy pattern to select the correct handler based on the operation type. Therefore, you can remove this check.